### PR TITLE
chore(deps): bump github.com/hashicorp/vault-client-go in /modules/vault

### PR DESCRIPTION
**Pull request by mmorel-35**
## What does this PR do?

Updates vault-client-go to 0.3.3

## Why is it important?

Stay up to date

### CODE REVIEW

**Comment by mdelapenya** [comment]
I think this is probably related: https://github.com/testcontainers/testcontainers-go/pull/1489#issuecomment-1672953015

Did the tests run and pass in your machine?

**Comment by mmorel-35** [comment]
> Did the tests run and pass in your machine?

Not Yet !
For the moment, I have 
```
.\vault_test.go:105:32: invalid operation: cannot index s.Data (variable of type schema.KvV2ReadResponse)
.\vault_test.go:128:31: invalid operation: cannot index s.Data (variable of type schema.KvV2ReadResponse)
```

File: modules/vault/vault_test.go [hunk 1]
```diff
@@ -37,6 +37,7 @@ func TestMain(m *testing.M) {
 		testcontainervault.WithToken(token),
 		// }
 		// WithInitCommand {
+		testcontainervault.WithInitCommand("secrets enable -version=2 kv"),
```
Comment (mmorel-35) [inline modules/vault/vault_test.go:40]: After reading [this](https://developer.hashicorp.com/vault/docs/secrets/kv/kv-v2#setup)

Comment (mmorel-35) [reply modules/vault/vault_test.go]: Doesn’t works...

**Comment by mmorel-35** [comment]
@mdelapenya , can you take it from here ?

**Comment by mmorel-35** [comment]
Hi @mdelapenya , 
The tests are passing now ;) !

**Review by mdelapenya — APPROVED** [review]
Great you found how to fix that error! I'm updating #1603 with these changes

Thanks!

**Pull request merged**